src/components/Explore/TimeSelector.tsx

```diff
@@ -0,0 +1,179 @@
+import { useOnClickOutside } from 'hooks/useOnClickOutside'
+import { useRef } from 'react'
+import React, { useState } from 'react'
+import { Check, ChevronDown, ChevronUp } from 'react-feather'
+import { useModalOpen, useToggleModal } from 'state/application/hooks'
+import { ApplicationModal } from 'state/application/reducer'
+import styled, { css } from 'styled-components/macro'
+
+const TIME_DISPLAYS: { [key: string]: string } = {
+  hour: '1H',
+  day: '1D',
+  week: '1W',
+  month: '1M',
+  year: '1Y',
+}
+
+const enum TimeOption {
+  Hour = 'hour',
+  Day = 'day',
+  Week = 'week',
+  Month = 'month',
+  Year = 'year',
+}
+const TIMES = [TimeOption.Hour, TimeOption.Day, TimeOption.Week, TimeOption.Month, TimeOption.Year]
+
+export enum FlyoutAlignment {
+  LEFT = 'LEFT',
+  RIGHT = 'RIGHT',
+}
+
+const InternalMenuItem = styled.div`
+  flex: 1;
+  padding: 0.5rem 0.5rem;
```

Comment:
We generally try to avoid `rem`, since it can have unexpected cascading size effects. Prefer `em` (or px values that are divisible by 4). I tend to do the % 4 px value closest to what's specified in Figma, and let the designer know if it's not perfectly divisible.